src/stack/Stack.java

```diff
@@ -1,56 +1,105 @@
 package stack;
 
-public class Stack {
+import java.util.Arrays;
 
-	static final int DEFAULT_SIZE = 16;
-	private int[] data;
-	private int tos; //index which determines the top of stack
+public class Stack {
+    public static void main(String[] args) {
+        Stack stack = new Stack(5);
+        System.out.println(stack);
+        stack.push(1);
+        stack.push(2);
+        stack.push(3);
+        stack.push(4);
+        stack.push(5);
+        stack.push(6);
+        stack.push(7);
+        System.out.println(stack);
+        stack.pop();
+        System.out.println(stack);
+        System.out.println(stack.isEmpty());
+        stack.clear();
+        System.out.println(stack);
 
-	public Stack() {
-		this(DEFAULT_SIZE);
-	}
 
-	public Stack(int size) {
-		data = new int[size];
-		this.tos = -1;
-	}
+    }
 
-	/**
-	 * Add element into stack
-	 */
-	public void push(int value) {
+    static final int DEFAULT_SIZE = 16;
+    private int[] data;
+    private int tos; //index which determines the top of stack
 
-	}
+    public Stack() {
+        this(DEFAULT_SIZE);
+    }
 
-	/**
-	 * Get element from the stack
-	 * @return
-	 */
-	public int pop() {
+    public Stack(int size) {
+        data = new int[size];
+        this.tos = 0;
+    }
 
-		return 0;
-	}
+    /**
+     * Add element into stack
+     */
+    public void push(int value) {
+        if (tos == data.length) {
```

Comment:
if (tos == data.length) in this case (tos==data.length-1) because  at first tos=-1;